The Crown Reactor and other changes, Ready to review [mapping changes to all maps]

Bod9001 opened the pull request:
# TODO 
* [x] ~~FIX Enumerators (on tab prefabs)~~
* [x] ~~add power component to stuff that requires it~~

# NOTES!!!
Requires scene changes for the new devices that require power to be hooked up
There might be some few accidental changes in here somewhere
Some of these features might be a bit buggy but I don't mind fixing them
Any changes needed just tell me

# Adds
* Periodic updates for update manager so you can specify how long the the delay between updatesare
* Multitool interface For setting stuff with the multitool and things like that
Made more devices use power

* netlabe now support text mesh pro
*supplying devices now can work on watts (  automatically doing the calculation to turn it into current and voltage)
* Adds radiation system (WIP, the resistivity of walls and such needs to set themselves on the meta data node still, but apart from that it seems to work well )
* Made it so underfloor and tile palette supports rotated and colour changed tiles (WIP, have to add the extra networking for it and change shaders on tile maps)
* reactor stuff
* Liquid pipes a bit WIP, (planning to add gas pipes after finished with reactor)
* Explosion manager
* Radiation does damage now
# what needs testing?
_Everything_
nah, main focus points is underfloor layer stuff and multitool changes

## Why?
Because I'm getting so far behind and holding up a few people with features that I've implemented here

so I've got a bunch of useful changes on my branch,
if I want to merge it all that would have to change would be wrapping up some of the machines I made powered
Stuff like reactors only matter if they're in the scene
I feel like the improvements have made will help With a few things
like Making a update manager that can do periodic updates with a specified time
So you don't have to do the enumerator yields stuff
Since apparently that generates long stack traces and gets slower and slower

thanks to @ThatDan123 for the finishing touches and mapping

Bod9001 committed:
Partially implemented reactor and suffer requires power

Bod9001 committed:
For fast compile times rebatings

Bod9001 committed:
Radiation yay, Needs optimisation but on its own thread so not too bad

Bod9001 committed:
Cleaned up for PR

Bod9001 committed:
ok, I'll actually try PRing it

Bod9001 committed:
Necessary changes and fixes

daalta commented on UnityProject/Assets/Scripts/Electricity/NodeModules/ResistanceSourceModule.cs:
```diff
@@ -125,7 +123,7 @@ public override void PowerNetworkUpdate()
 		if (dirtyResistance)
 		{
-			if (NotEditorResistanceset)
+			if (NotEditorResistanceset )
```
Spess

daalta commented on UnityProject/Assets/Scripts/Construction/ConveyorBelts/ConveyorBelt.cs:
```diff
@@ -317,4 +317,18 @@ public void ServerPerformInteraction(HandApply interaction)
 		}
 	}
+
+	//######################################## Multitool interaction ##################################
```
Might as well make this a `Header["Multitool interaction"]`

Bod9001 replied:
How does it work exactly?

daalta replied:
Oh, headers are just text that appears in the inspector, like a headline. 

So you'll see that the following serialized variables are for "Multitool interaction", but it'll also be visible in the inspector 🙂

https://docs.unity3d.com/ScriptReference/HeaderAttribute.html

Bod9001 replied:
oh ok, thought it was some type of code separation stuff

Bod9001 committed:
update

Bod9001 committed:
Fixed

ThatDan123 commented on UnityProject/Assets/Scripts/Electricity/Engine/BoilerTurbineController.cs:
```diff
@@ -0,0 +1,43 @@
+	public ReactorBoiler ReactorBoiler = null;
+	public ReactorTurbine ReactorTurbine = null;
+    // Start is called before the first frame update
+    void Start()
```
Unneeded start method

ThatDan123 commented on UnityProject/Assets/Scripts/Construction/ConveyorBelts/ConveyorBelt.cs:
```diff
@@ -317,4 +317,19 @@ public void ServerPerformInteraction(HandApply interaction)
+	private MultitoolConnectionType conType = MultitoolConnectionType.Conveyor;
+	public MultitoolConnectionType ConType  => conType;
+
+	//[SerializeField]
```
Supposed to be commented out?

Bod9001 replied:
yeah Unity was being Dumdum with Changing the values in the code but not updating on the serialised inspector

ThatDan123 commented on UnityProject/Assets/Scripts/Electricity/Engine/GUI_ReactorControler.cs:
```diff
@@ -0,0 +1,320 @@
+	}
+
+	/// <summary> As
+	/// Sets shuttle speed.
```
Wrong summary??

ThatDan123 commented on UnityProject/Assets/Scripts/Electricity/Engine/PowerControlConsole.cs:
```diff
@@ -0,0 +1,32 @@
+
+	private float PreviousVoltage = 0;
+
+	void IAPCPowered.PowerNetworkUpdate(float Voltage)
```
needed? Whole method internals commented out.

ThatDan123 replied:
```
{
		/*ElectricityFunctions.WorkOutActualNumbers(ThisDevice.ControllingNode.Node.InData);
		var ControllerMultiplier = (TargetsSupplyOutputVoltage+1) / (ThisDevice.ControllingNode.Node.InData.Data.ActualVoltage+1);
		Logger.Log("in ControllerMultiplier " + ControllerMultiplier+"  >> " +  (TargetsSupplyOutputVoltage+1)+ " / " + (ThisDevice.ControllingNode.Node.InData.Data.ActualVoltage+1));
		if (ControllerMultiplier != 0 && !float.IsInfinity(ControllerMultiplier))
		{
			ControlConsole.RequestRelativeChange(ControllerMultiplier);
		}
		PreviousVoltage = ThisDevice.ControllingNode.Node.InData.Data.ActualVoltage;*/

	}
```

Bod9001 replied:
yeah, It's not used for anything currently was meant for automatically increasing the output of the reactor but is a bit more tricky to calculate then I realised Should I remove it?

ThatDan123 replied:
Guess it depends on whether you want to come back and try again, no harm keeping it in if thats the case, though maybe add a comment on it just so you dont forget why its commented out :p

ThatDan123 commented on UnityProject/Assets/Scripts/Electricity/Engine/ReactorControlConsole.cs:
```diff
@@ -0,0 +1,61 @@
+
+		ReactorChambers.ControlRodDepthPercentage *= (1 / Multiplier);
+		ControlRodDepthPercentage = ReactorChambers.ControlRodDepthPercentage;
+//Change this to use SetControlRodDepth
```
Does this need doing, also are these logs needed?

ThatDan123 commented on UnityProject/Assets/Scripts/Electricity/NodeModules/ModuleSupplyingDevice.cs:
```diff
@@ -69,13 +79,15 @@ public override void OnDespawnServer(DespawnInfo info)
+	[RightClickMethod]
 	public override void TurnOnSupply()
 	{
+		Logger.Log("TurnOnSupply");
```
Log needed?

ThatDan123 commented on UnityProject/Assets/Scripts/Input System/InteractionV2/TileInteraction/PipeDeconstruction.cs:
```diff
@@ -0,0 +1,64 @@
+		var metaDataNode = matrix.GetMetaDataNode(interaction.TargetCellPos);
+		for (var i = 0; i < metaDataNode.PipeData.Count; i++)
+		{
+			Logger.Log("OH?");
```
Logs needed?

ThatDan123 commented on UnityProject/Assets/Scripts/Radiation/RadiationProducer.cs:
```diff
@@ -0,0 +1,99 @@
+		}
+
+		// Update is called once per frame
+		void Update()
```
empty update method

ThatDan123 commented on UnityProject/Assets/Scripts/Radiation/RadiationProducer.cs:
```diff
@@ -0,0 +1,99 @@
+		}
+
+		// Start is called before the first frame update
+		void Start()
```
empty start method

ThatDan123 commented on UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/MetaDataSystem.cs:
```diff
@@ -211,11 +211,11 @@ private void SetupNeighbors(MetaDataNode node)
-				// if current node is a room, but the neighboring is a space tile, this node needs to be checked regularly for changes by other matrices
-				if (node.IsRoom && !externalNodes.ContainsKey(node))
+				/*// if current node is a room, but the neighboring is a space tile, this node needs to be checked regularly for changes by other matrices
+				if (node.IsRoom && !externalNodes.ContainsKey(node) && metaTileMap.IsSpaceAt(node.Position, true) == false)
```
This supposed to be commented out?

Bod9001 replied:
I was having lag issues with the explosion, Turns out if I commenting this out Solve the lag issue And it seems to not impact anything

ThatDan123 commented on UnityProject/Assets/Scripts/UI/Net/Elements/NetSlider.cs:
```diff
@@ -24,7 +24,7 @@ public override string Value
 	public FloatEvent ServerMethod;
 
-	private Slider element;
+	public Slider element;
```
Needs to be public?

Bod9001 committed:
Fixed stuff

Bod9001 committed:
Forgot something

ReiderJack commented on UnityProject/Assets/Scripts/Electricity/Electrical processes/ElectricalSynchronisation.cs:
```diff
@@ -380,7 +382,7 @@ private void PowerUpdateResistanceChange()
 		for (int i = ResistanceChange.Count - 1; i >= 0; i--)
 		{
-			if (i < InitialiseResistanceChange.Count)
+			if (i < ResistanceChange.Count)
```
redundant

Bod9001 committed:
Simplify stuff and fix an error

Bod9001 committed:
Remove redundant consul

Bod9001 committed:
Requested changes

Bod9001 committed:
added Sprites

Bod9001 committed:
Merge branch 'develop' into Electrical-gameplay

Bod9001 committed:
Merge branch 'develop' into Electrical-gameplay

Bod9001 committed:
Tweaked  boiler

Bod9001 committed:
Merge branch 'Electrical-gameplay' of https://github.com/Bod9001/unitystation into Electrical-gameplay

Bod9001 committed:
Outpost done

ThatDan123 committed:
Merge pull request #2 from Bod9001/Electrical-gameplay

Outpost done

ThatDan123 committed:
Boxstation APC references

ThatDan123 committed:
Relicta APC references

ThatDan123 committed:
Temple APC references

ThatDan123 committed:
Centcom APC references

ThatDan123 committed:
Fallstation APC references

ThatDan123 committed:
Pogstation APC references

Bod9001 committed:
Merge pull request #1 from ThatDan123/pr/4358

Adds the APC references to all maps apart from outpost which was done

ThatDan123 committed:
Merge remote-tracking branch 'upstream/develop' into pr/4358

ThatDan123 committed:
Proper Fallstation APC reference fix

Bod9001 committed:
Merge pull request #2 from ThatDan123/pr/4358

Fixes fallstation and updates to latest.

Bod9001 committed:
removes The editor generation of light meshes,

There are only needed in runtime and also clog up the change list on the scenes

Bod9001 committed:
Merge branch 'Electrical-gameplay' of https://github.com/Bod9001/unitystation into Electrical-gameplay

Bod9001 committed:
Tiny fix

Bod9001 committed:
Fixes rotation issue

Bod9001 committed:
Gateways work now

ThatDan123 committed:
Box reactor

Bod9001 committed:
Merge pull request #3 from ThatDan123/pr/4358

Boxstation reactor

Bod9001 committed:
Fixes and pog mapping

Bod9001 committed:
Merge branch 'Electrical-gameplay' of https://github.com/Bod9001/unitystation into Electrical-gameplay

Bod9001 committed:
Makes it so you can't stack pipes on top of each other

Bod9001 committed:
added Radiation protection and radiation pills

Bod9001 committed:
Added radiation protection to outpost

ThatDan123 committed:
Radiation Locker Contents

Bod9001 committed:
added Geiger counter to Lockers

ThatDan123 committed:
Cargo stuff for rods.

ThatDan123 committed:
Prefab name and export stuff

ThatDan123 committed:
Merge branch 'Electrical-gameplay' into pr/4358

ThatDan123 committed:
Names and descriptions for the other reactor objects

Bod9001 committed:
Merge pull request #4 from ThatDan123/pr/4358

Cargo and prefab names

ThatDan123 committed:
Box station additions

ThatDan123 committed:
Fix blast door check

Bod9001 committed:
Merge pull request #5 from ThatDan123/pr/4358

Adds more blast doors and radiation medical stuff to box

Bod9001 committed:
Basic geiger counter

ThatDan123 committed:
Pog fixes

Bod9001 committed:
Merge pull request #6 from ThatDan123/pr/4358

Pog fixes

Bod9001 committed:
Geiger counter fixed and reactor fixed

Bod9001 committed:
Merge branch 'Electrical-gameplay' of https://github.com/Bod9001/unitystation into Electrical-gameplay
